### PR TITLE
Drop size 0 arrays in stack

### PR DESCRIPTION
**Pull request by jakirkham**
Fixes https://github.com/dask/dask/issues/4928

Was doing some cleanup of `da.stack` ( https://github.com/dask/dask/pull/4975 ) ( https://github.com/dask/dask/pull/4976 ) after [the recent blogpost]( https://blog.dask.org/2019/06/20/load-image-data ) (since we reference using `da.stack` there). Was curious to see what it would take to solve issue ( https://github.com/dask/dask/issues/4928 ) and found it was pretty easy. Anyways here is the fix.

Note: Requires PR ( https://github.com/dask/dask/pull/4976 ) be merged first.

- [x] Tests added / passed
- [x] Passes `flake8 dask`

### CODE REVIEW

**Comment by mrocklin** [comment]
Thoughts on adding a small test?

**Comment by jakirkham** [comment]
Happy to take a look. Given this is waiting on PR ( https://github.com/dask/dask/pull/4976 ) to be merged first, will hold off until that happens.

**Comment by jakirkham** [comment]
I've pushed a test here that I think should work. It relies on some dev versions though.

**Comment by jakirkham** [comment]
Also added a test that works with stable versions.

**Comment by jakirkham** [comment]
Ok should be good now. If we can include this in this release, that would be nice. Though if that's not possible, that's ok.

**Comment by mrocklin** [comment]
Is this critical for something?  Ideally it's nice for PRs to be in master
for a bit to flush out the possibility of errors.  Merging things just
before release makes me a little nervous.

**Comment by jakirkham** [comment]
Just to reiterate, it's up to you to merge or not. Am not trying to for the issue. 🙂

File: dask/array/tests/test_sparse.py [hunk 1]
```diff
@@ -168,6 +168,8 @@ def test_metadata():
         assert isinstance(np.concatenate([y, y])._meta, sparse.COO)
         assert isinstance(np.concatenate([y, y[:0], y])._meta, sparse.COO)
         assert isinstance(np.stack([y, y])._meta, sparse.COO)
+        if np.__version__ >= "1.17":
```
Comment (TomAugspurger) [inline dask/array/tests/test_sparse.py]: Can I bother you to make a `_numpy_117` variable in `dask/array/numpy_compat.py` and use that here? I find it easier to clean things up down the road when all the versions-specific things refer to variables.

Comment (jakirkham) [reply dask/array/tests/test_sparse.py]: Sounds reasonable Tom. Have pushed such a change. Please check and see if this addresses your concern.

**Comment by jakirkham** [comment]
Planning to merge end of day if no comments.

File: dask/array/tests/test_sparse.py [hunk 2]
```diff
@@ -168,6 +169,8 @@ def test_metadata():
         assert isinstance(np.concatenate([y, y])._meta, sparse.COO)
         assert isinstance(np.concatenate([y, y[:0], y])._meta, sparse.COO)
         assert isinstance(np.stack([y, y])._meta, sparse.COO)
+        if np.__version__ >= _numpy_117:
```
Comment (jrbourbeau) [inline dask/array/tests/test_sparse.py]:
```suggestion
        if _numpy_117:
```

Comment (jakirkham) [reply dask/array/tests/test_sparse.py]: Oops. Thanks for catching that James.

**Comment by jakirkham** [comment]
Thanks all! 😄

**Pull request merged**